packages/react/src/constants/device-sizes.constant.ts

```diff
@@ -36,3 +38,42 @@ export const sizes = {
     return 'large';
   },
 };
+
+interface Breakpoints {
+  small?: number;
+  medium?: number;
+}
+
+export const getSizesForBreakpoints = ({ small, medium }: Breakpoints) => {
+  const newSizes = {
+    ...sizes, // Note: this helps get the function from sizes
+    ...fastClone(sizes), // Note: this helps to get a deep clone of fields like small, medium etc
+  };
+
+  if (!small || !medium) {
+    return newSizes;
+  }
+
+  newSizes.small = {
+    max: small,
+    min: Math.floor(small / 2),
+    default: 0, // Note: For TS
+  };
+  newSizes.small.default = newSizes.small.min + 1; // TODO: handle negative values?
+
+  newSizes.medium = {
+    max: medium,
+    min: newSizes.small.max + 1,
+    default: 0, // Note: For TS
+  };
+  newSizes.medium.default = newSizes.medium.min + 1; // TODO: handle negative values?
+
+  newSizes.large = {
+    max: 2000, // TODO: decide upper limit
+    min: newSizes.medium.max + 1,
+    default: 0, // Note: For TS
```

Comment:
so we don't need to use the `default` values at all?

Reply:
I see that `default` is not being used in the SDK anywhere. In that case, let's just remove it from the new sizes, the old sizes, and from the types. Otherwise its presence will cause confusion

Reply:
Are you sure we can set these to `0`? I am looking at where `sizes` gets used: it gets fed to `this.state.state` which is provided to every custom component and custom code block that is evaluated.

This makes me worry that we cannot set this to `0`. We should probably stick with the old way of setting it to be safe

Reply:
done, as per our conversation